subs/pantry/src/Pantry/Types.hs

```diff
@@ -437,6 +438,15 @@ instance Display PantryException where
     ":\n" <>
     displayShow e
 
+data PantryDownloadException = BadKey !Text
```

Comment:
I'd recommend just putting this in with `PantryDownloadException` as another data constructor.

Reply:
Do you mean as another constructor into `PantryException`?

Reply:
Yes